crates/solidity/inputs/language/bindings/rules.msgb

```diff
@@ -2060,15 +2060,48 @@ inherit .star_extension
   edge @enum.def -> type
   edge type -> type_enum_type
   edge type_enum_type -> @enum.lexical_scope
-}
 
-@enum [EnumDefinition
+  ; Path to link min and max to the enum's scope (#1158)
+  ; It resolves paths of the form `type(<enum>).min.<something>`
+  node typeof
+  attr (typeof) pop_symbol = "@typeof"
+
+  node built_in_member
+  attr (built_in_member) pop_symbol = "."
+  node min_built_in
+  attr (min_built_in) pop_symbol = "min"
+  node max_built_in
+  attr (max_built_in) pop_symbol = "max"
+  node minmax_type_of
+  attr (minmax_type_of) push_symbol = "@typeof"
+  node minmax_replace
+  attr (minmax_replace) push_symbol = (source-text @name)
+
+  edge type -> typeof
+  edge typeof -> built_in_member
+  edge built_in_member -> min_built_in
+  edge built_in_member -> max_built_in
+  edge min_built_in -> minmax_type_of
+  edge max_built_in -> minmax_type_of
+  edge minmax_type_of -> minmax_replace
+  edge minmax_replace -> @enum.lexical_scope
+}
+
+@enum [EnumDefinition @name name: [Identifier]
     members: [EnumMembers @item [Identifier]]
 ] {
   node def
   attr (def) node_definition = @item
   attr (def) definiens_node = @item
 
+  node type_of
+  attr (type_of) push_symbol = "@typeof"
+  node type_def
+  attr (type_def) push_symbol = (source-text @name)
+  edge def -> type_of
+  edge type_of -> type_def
+  edge type_def -> @enum.lexical_scope
```

Comment:
Here instead of repeating the path segment you can use the aforementioned path and do only add:
```
edge def -> @enum.value_ref_typeof`
```
You'll not need to capture the enum's name either.